base/irrationals.jl

```diff
@@ -158,6 +158,10 @@ for op in Symbol[:+, :-, :*, :/, :^]
     @eval $op(x::AbstractIrrational, y::AbstractIrrational) = $op(Float64(x),Float64(y))
 end
 *(x::Bool, y::AbstractIrrational) = ifelse(x, Float64(y), 0.0)
+function ^(x::AbstractIrrational, y::Integer)
```

Comment:
I would argue that if `y` is a `BigInt`, it should use `BigFloat` instead of `Float64`.

Reply:
That was my initial idea as well (given that that's what the promotion mechanism `^(promote(x,y)...)` will produce), however that doesn't fit into the `power_by_squaring` framework which is used by literal integer exponents. I've tried to keep the present behavior unchanged while being consistent.

Reply:
I think this should use `BigFloat` based on the logic that any number expresable as a `Float64` raised to a power >2^62 will need BigBloat to store the result.

Reply:
Would the best idea then be to use the promotion mechanism, and define the exponentiation as `^(promote(x,y)...)`?

Reply:
I wouldn’t use `promote`, since it would turn the integer exponent into a floating-point number.